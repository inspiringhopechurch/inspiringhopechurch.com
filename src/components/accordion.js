import React, { useState, useEffect } from "react";
import * as PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import "./accordion.sass";

export const invalidChars = /^[^a-zA-Z]+|[^\w:.-]+/g;

const Accordion = ({ title, children, isExpanded }) => {
  const [expanded, setExpanded] = useState(isExpanded);
  const [accordionHeight, setAccordionHeight] = useState(0);

  const toggleAccordion = () => {
    expanded && accordionContent && accordionContent.removeAttribute("style");
    setExpanded(!expanded);
  };

  useEffect(() => {
    // Instead of resetting state when we resize, just remove styles, because this is simpler.
    function handleWindowResize() {
      // TODO: debounce removing style attribute. 1 operation/sec should be sufficient
      accordionContent.removeAttribute("style");
    }

    //! NOTE: DOM Event listeners set 'this' to be the target element,
    //! and if you rely on 'this' in an event handler, a regular function is necessary
    window.addEventListener("resize", handleWindowResize);

    return function cleanup() {
      window.removeEventListener("resize", handleWindowResize);
    };
  });

  const accordionId = title.replace(invalidChars, "-");

  // Check for document being available first to prevent error during
  // gatsby ssr build step: https://www.gatsbyjs.com/docs/debugging-html-builds/
  const accordionContent = typeof document !== `undefined` && document.querySelector(`[data-id='${accordionId}']`);

  // We make sure to not reset accordionHeight to a value that's less than its initial value.
  // It will be on static pages whose data will not change, so there should never be 'less'
  // content that what we start with.
  // The exception to this is when we move from smaller viewports to wider ones (height will be smaller).
  // FOr that scenario we have the useEffect above dealing with window resizing.
  if (accordionContent?.clientHeight && accordionContent?.clientHeight > accordionHeight) {
    setAccordionHeight(accordionContent.clientHeight);
  }

  return (
    <>
      <h2 className={`accordion is-size-4 is-uppercase${expanded ? " expanded" : ""}`}>
        <button aria-expanded={expanded ? "true" : "false"} onClick={toggleAccordion}>
          {title} <FontAwesomeIcon icon={["fas", expanded ? "minus" : "plus"]} size="sm" />
        </button>
      </h2>
      <div
        className="accordion-content"
        data-id={accordionId} // @ts-ignore custom css property "--accordion-max-height"
        style={expanded && accordionHeight > 1 ? { "--accordion-max-height": `${accordionHeight}px` } : {}}
      >
        {children}
      </div>
    </>
  );
};

Accordion.propTypes = {
  title: PropTypes.string.isRequired,
};

Accordion.defaultProps = {
  isExpanded: false,
};

export default Accordion;
