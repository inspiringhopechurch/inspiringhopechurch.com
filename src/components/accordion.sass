@charset "utf-8"

@import "../styles/_myvariables.sass"

.accordion
    > button
        all: inherit
        width: 100%
        position: relative
        cursor: pointer
        &:before, &:after
            content: ""
            position: absolute
            background-color: $link
            transition: transform $accordion-transition-time ease-out
        // Vertical line
        &:before
            top: 0
            right: 0.6875rem
            width: .25rem
            height: 1.625rem
        // horizontal line
        &:after
            top: .6875rem
            right: 0
            width: 1.625rem
            height: .25rem

    &.expanded > button
        &:before
            transform: rotate(90deg)
        &:after
            transform: rotate(180deg)

    &.expanded + .accordion-content
        max-height: var( --accordion-max-height )
        opacity: 1
        transition: all $accordion-transition-time ease-in
    & + .accordion-content
        max-height: 0
        opacity: 0
        overflow: hidden
        transition: all $accordion-transition-time ease-out
