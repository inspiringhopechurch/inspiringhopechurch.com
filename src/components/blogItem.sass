@charset "utf-8"

@import "../styles/_myvariables.sass"
@import "bulma/sass/elements/button.sass"

.blog-info-container
    position: relative
    text-align: center
    
.image.blog-image
    margin: 0
    
.blog-info
    top: -5rem
    position: relative
    width: 85%
    margin: 0 auto
    margin-bottom: -3rem
    box-shadow: 0 0.5em 1em -0.125em rgba($scheme-invert, 0.1), 0 0px 0 1px rgba($scheme-invert, 0.02)
    background-color: $white
    padding: 2rem 1rem
    @include from($tablet)
        padding: 2rem
    
.blog-excerpt
    text-align: initial
    @include from($tablet)
        padding: 0 1rem
.button
    transition: background $decoration-transition-time
