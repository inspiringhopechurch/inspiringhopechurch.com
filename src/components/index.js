import Accordion from "../components/accordion";
import BlogItem from "./blogItem";
import ContactForm from "./contactForm";
import FancyHeading from "./fancyHeading";
import { Map } from "./map";
import MediaItem from "../components/mediaItem";
import Pagination from "./pagination";
import { RefTagger } from "./reftagger";
import SEO from "./seo";

export { Accordion, BlogItem, ContactForm, FancyHeading, Map, MediaItem, Pagination, RefTagger, SEO }