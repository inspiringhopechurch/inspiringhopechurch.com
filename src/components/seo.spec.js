import React from "react";
import { mount } from "enzyme";
import { SEO } from "../components";
import { useStaticQuery } from "gatsby";
import data from "../fixtures/data";

describe("SEO", () => {
  beforeEach(() => {
    const buildTime_ = data.data.site.buildTime,
      siteMetaD = data.data.site.siteMetadata;
    // mock StaticQuery for tests
    useStaticQuery.mockImplementationOnce(({ render }) =>
      render({
        site: {
          buildTime: buildTime_,
          siteMetadata: {
            defaultTitle: siteMetaD.defaultTitle,
            titleAlt: siteMetaD.titleAlt,
            shortName: siteMetaD.shortName,
            author: siteMetaD.author,
            siteLanguage: siteMetaD.siteLanguage,
            logo: siteMetaD.logo,
            siteUrl: siteMetaD.siteUrl,
            url: siteMetaD.url,
            pathPrefix: siteMetaD.pathPrefix,
            defaultDescription: siteMetaD.defaultDescription,
            defaultBanner: siteMetaD.defaultBanner,
            twitter: siteMetaD.twitter
          }
        }
      })
    );
  });

  it("should render without crashing", () => {
    const mySEO = mount(<SEO />);
    expect(mySEO).toMatchSnapshot();
  });

  it("should render without crashing as an article", () => {
    const mySEO = mount(<SEO article />);
    expect(mySEO).toMatchSnapshot();
  });

  it("should render without crashing with custom props", () => {
    const mySEO = mount(
      <SEO
        title="Test Page"
        desc={"This website is an awesome website and you should visit it."}
        pathname={"/path/to/page"}
        banner={"/path/to/image.jpg"}
      />
    );
    expect(mySEO).toMatchSnapshot();
  });

  it("should render without crashing as an article with custom props", () => {
    const mySEO = mount(
      <SEO
        article
        title="Test Page"
        desc={"This website is an awesome website and you should visit it."}
        pathname={"/path/to/page"}
        banner={"/path/to/image.jpg"}
      />
    );
    expect(mySEO).toMatchSnapshot();
  });
});
