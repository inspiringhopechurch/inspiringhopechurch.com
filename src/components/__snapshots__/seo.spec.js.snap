// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEO should render without crashing 1`] = `
<SEO
  article={false}
  banner={null}
  desc={null}
  pathname={null}
  title={null}
>
  <mockConstructor
    render={[Function]}
  >
    <HelmetWrapper
      defer={false}
      encodeSpecialCharacters={true}
      title="Test Company Site"
    >
      <SideEffect(NullComponent)
        defer={false}
        encodeSpecialCharacters={true}
        htmlAttributes={
          Object {
            "class": "has-navbar-fixed-top",
            "lang": "en",
          }
        }
        link={
          Array [
            Object {
              "href": "https://inspiringhopechurch.com/",
              "rel": "canonical",
            },
          ]
        }
        meta={
          Array [
            Object {
              "content": "We exist to inspire hope by helping people discover God’s purpose for their lives through a relationship with Jesus.",
              "name": "description",
            },
            Object {
              "content": "https://inspiringhopechurch.com/src/assets/logo.svg",
              "name": "image",
            },
            Object {
              "content": "TestCompany",
              "name": "apple-mobile-web-app-title",
            },
            Object {
              "content": "TestCompany",
              "name": "application-name",
            },
            Object {
              "content": "Test Company Site",
              "property": "og:site_name",
            },
            Object {
              "content": "https://inspiringhopechurch.com/",
              "property": "og:url",
            },
            Object {
              "content": "website",
              "property": "og:type",
            },
            Object {
              "content": "Test Company Site",
              "property": "og:title",
            },
            Object {
              "content": "We exist to inspire hope by helping people discover God’s purpose for their lives through a relationship with Jesus.",
              "property": "og:description",
            },
            Object {
              "content": "https://inspiringhopechurch.com/src/assets/logo.svg",
              "property": "og:image",
            },
            Object {
              "content": "summary_large_image",
              "name": "twitter:card",
            },
            Object {
              "content": "Test Company Site",
              "name": "twitter:title",
            },
            Object {
              "content": "We exist to inspire hope by helping people discover God’s purpose for their lives through a relationship with Jesus.",
              "name": "twitter:description",
            },
            Object {
              "content": "https://inspiringhopechurch.com/src/assets/logo.svg",
              "name": "twitter:image",
            },
            Object {
              "content": "https://inspiringhopechurch.com/",
              "name": "twitter:url",
            },
            Object {
              "content": "@inspiringhope",
              "name": "twitter:creator",
            },
            Object {
              "content": "https://twitter.com/inspiringhope/",
              "name": "twitter:site",
            },
          ]
        }
        script={
          Array [
            Object {
              "innerHTML": "[{\\"@context\\":\\"http://schema.org\\",\\"@type\\":\\"WebSite\\",\\"@id\\":\\"https://inspiringhopechurch.com\\",\\"url\\":\\"https://inspiringhopechurch.com\\",\\"name\\":\\"Test Company Site\\",\\"alternateName\\":\\"Test Company Site\\"}]",
              "type": "application/ld+json",
            },
          ]
        }
        title="Test Company Site"
      >
        <NullComponent
          defer={false}
          encodeSpecialCharacters={true}
          htmlAttributes={
            Object {
              "class": "has-navbar-fixed-top",
              "lang": "en",
            }
          }
          link={
            Array [
              Object {
                "href": "https://inspiringhopechurch.com/",
                "rel": "canonical",
              },
            ]
          }
          meta={
            Array [
              Object {
                "content": "We exist to inspire hope by helping people discover God’s purpose for their lives through a relationship with Jesus.",
                "name": "description",
              },
              Object {
                "content": "https://inspiringhopechurch.com/src/assets/logo.svg",
                "name": "image",
              },
              Object {
                "content": "TestCompany",
                "name": "apple-mobile-web-app-title",
              },
              Object {
                "content": "TestCompany",
                "name": "application-name",
              },
              Object {
                "content": "Test Company Site",
                "property": "og:site_name",
              },
              Object {
                "content": "https://inspiringhopechurch.com/",
                "property": "og:url",
              },
              Object {
                "content": "website",
                "property": "og:type",
              },
              Object {
                "content": "Test Company Site",
                "property": "og:title",
              },
              Object {
                "content": "We exist to inspire hope by helping people discover God’s purpose for their lives through a relationship with Jesus.",
                "property": "og:description",
              },
              Object {
                "content": "https://inspiringhopechurch.com/src/assets/logo.svg",
                "property": "og:image",
              },
              Object {
                "content": "summary_large_image",
                "name": "twitter:card",
              },
              Object {
                "content": "Test Company Site",
                "name": "twitter:title",
              },
              Object {
                "content": "We exist to inspire hope by helping people discover God’s purpose for their lives through a relationship with Jesus.",
                "name": "twitter:description",
              },
              Object {
                "content": "https://inspiringhopechurch.com/src/assets/logo.svg",
                "name": "twitter:image",
              },
              Object {
                "content": "https://inspiringhopechurch.com/",
                "name": "twitter:url",
              },
              Object {
                "content": "@inspiringhope",
                "name": "twitter:creator",
              },
              Object {
                "content": "https://twitter.com/inspiringhope/",
                "name": "twitter:site",
              },
            ]
          }
          script={
            Array [
              Object {
                "innerHTML": "[{\\"@context\\":\\"http://schema.org\\",\\"@type\\":\\"WebSite\\",\\"@id\\":\\"https://inspiringhopechurch.com\\",\\"url\\":\\"https://inspiringhopechurch.com\\",\\"name\\":\\"Test Company Site\\",\\"alternateName\\":\\"Test Company Site\\"}]",
                "type": "application/ld+json",
              },
            ]
          }
          title="Test Company Site"
        />
      </SideEffect(NullComponent)>
    </HelmetWrapper>
  </mockConstructor>
</SEO>
`;

exports[`SEO should render without crashing as an article 1`] = `
<SEO
  article={true}
  banner={null}
  desc={null}
  pathname={null}
  title={null}
>
  <mockConstructor
    render={[Function]}
  >
    <HelmetWrapper
      defer={false}
      encodeSpecialCharacters={true}
      title="Test Company Site"
    >
      <SideEffect(NullComponent)
        defer={false}
        encodeSpecialCharacters={true}
        htmlAttributes={
          Object {
            "class": "has-navbar-fixed-top",
            "lang": "en",
          }
        }
        link={
          Array [
            Object {
              "href": "https://inspiringhopechurch.com/blog/",
              "rel": "canonical",
            },
          ]
        }
        meta={
          Array [
            Object {
              "content": "We exist to inspire hope by helping people discover God’s purpose for their lives through a relationship with Jesus.",
              "name": "description",
            },
            Object {
              "content": "https://inspiringhopechurch.com/src/assets/logo.svg",
              "name": "image",
            },
            Object {
              "content": "TestCompany",
              "name": "apple-mobile-web-app-title",
            },
            Object {
              "content": "TestCompany",
              "name": "application-name",
            },
            Object {
              "content": "Test Company Site",
              "property": "og:site_name",
            },
            Object {
              "content": "https://inspiringhopechurch.com/blog/",
              "property": "og:url",
            },
            Object {
              "content": "article",
              "property": "og:type",
            },
            Object {
              "content": "Test Company Site",
              "property": "og:title",
            },
            Object {
              "content": "We exist to inspire hope by helping people discover God’s purpose for their lives through a relationship with Jesus.",
              "property": "og:description",
            },
            Object {
              "content": "https://inspiringhopechurch.com/src/assets/logo.svg",
              "property": "og:image",
            },
            Object {
              "content": "summary_large_image",
              "name": "twitter:card",
            },
            Object {
              "content": "Test Company Site",
              "name": "twitter:title",
            },
            Object {
              "content": "We exist to inspire hope by helping people discover God’s purpose for their lives through a relationship with Jesus.",
              "name": "twitter:description",
            },
            Object {
              "content": "https://inspiringhopechurch.com/src/assets/logo.svg",
              "name": "twitter:image",
            },
            Object {
              "content": "https://inspiringhopechurch.com/blog/",
              "name": "twitter:url",
            },
            Object {
              "content": "@inspiringhope",
              "name": "twitter:creator",
            },
            Object {
              "content": "https://twitter.com/inspiringhope/",
              "name": "twitter:site",
            },
          ]
        }
        script={
          Array [
            Object {
              "innerHTML": "[{\\"@context\\":\\"http://schema.org\\",\\"@type\\":\\"BlogPosting\\",\\"@id\\":\\"https://inspiringhopechurch.com/blog/\\",\\"url\\":\\"https://inspiringhopechurch.com/blog/\\",\\"name\\":null,\\"alternateName\\":\\"Test Company Site\\",\\"headline\\":null,\\"image\\":{\\"@type\\":\\"ImageObject\\",\\"url\\":\\"https://inspiringhopechurch.com/src/assets/logo.svg\\"},\\"description\\":\\"We exist to inspire hope by helping people discover God’s purpose for their lives through a relationship with Jesus.\\",\\"datePublished\\":\\"2019-05-09\\",\\"dateModified\\":\\"2019-05-09\\",\\"author\\":{\\"@type\\":\\"Person\\",\\"name\\":\\"Test User\\"},\\"publisher\\":{\\"@type\\":\\"Organization\\",\\"name\\":\\"Test User\\",\\"logo\\":{\\"@type\\":\\"ImageObject\\",\\"url\\":\\"https://inspiringhopechurch.com/src/assets/logo.svg\\"}},\\"isPartOf\\":\\"https://inspiringhopechurch.com\\",\\"mainEntityOfPage\\":{\\"@type\\":\\"WebSite\\",\\"@id\\":\\"https://inspiringhopechurch.com\\"}}]",
              "type": "application/ld+json",
            },
          ]
        }
        title="Test Company Site"
      >
        <NullComponent
          defer={false}
          encodeSpecialCharacters={true}
          htmlAttributes={
            Object {
              "class": "has-navbar-fixed-top",
              "lang": "en",
            }
          }
          link={
            Array [
              Object {
                "href": "https://inspiringhopechurch.com/blog/",
                "rel": "canonical",
              },
            ]
          }
          meta={
            Array [
              Object {
                "content": "We exist to inspire hope by helping people discover God’s purpose for their lives through a relationship with Jesus.",
                "name": "description",
              },
              Object {
                "content": "https://inspiringhopechurch.com/src/assets/logo.svg",
                "name": "image",
              },
              Object {
                "content": "TestCompany",
                "name": "apple-mobile-web-app-title",
              },
              Object {
                "content": "TestCompany",
                "name": "application-name",
              },
              Object {
                "content": "Test Company Site",
                "property": "og:site_name",
              },
              Object {
                "content": "https://inspiringhopechurch.com/blog/",
                "property": "og:url",
              },
              Object {
                "content": "article",
                "property": "og:type",
              },
              Object {
                "content": "Test Company Site",
                "property": "og:title",
              },
              Object {
                "content": "We exist to inspire hope by helping people discover God’s purpose for their lives through a relationship with Jesus.",
                "property": "og:description",
              },
              Object {
                "content": "https://inspiringhopechurch.com/src/assets/logo.svg",
                "property": "og:image",
              },
              Object {
                "content": "summary_large_image",
                "name": "twitter:card",
              },
              Object {
                "content": "Test Company Site",
                "name": "twitter:title",
              },
              Object {
                "content": "We exist to inspire hope by helping people discover God’s purpose for their lives through a relationship with Jesus.",
                "name": "twitter:description",
              },
              Object {
                "content": "https://inspiringhopechurch.com/src/assets/logo.svg",
                "name": "twitter:image",
              },
              Object {
                "content": "https://inspiringhopechurch.com/blog/",
                "name": "twitter:url",
              },
              Object {
                "content": "@inspiringhope",
                "name": "twitter:creator",
              },
              Object {
                "content": "https://twitter.com/inspiringhope/",
                "name": "twitter:site",
              },
            ]
          }
          script={
            Array [
              Object {
                "innerHTML": "[{\\"@context\\":\\"http://schema.org\\",\\"@type\\":\\"BlogPosting\\",\\"@id\\":\\"https://inspiringhopechurch.com/blog/\\",\\"url\\":\\"https://inspiringhopechurch.com/blog/\\",\\"name\\":null,\\"alternateName\\":\\"Test Company Site\\",\\"headline\\":null,\\"image\\":{\\"@type\\":\\"ImageObject\\",\\"url\\":\\"https://inspiringhopechurch.com/src/assets/logo.svg\\"},\\"description\\":\\"We exist to inspire hope by helping people discover God’s purpose for their lives through a relationship with Jesus.\\",\\"datePublished\\":\\"2019-05-09\\",\\"dateModified\\":\\"2019-05-09\\",\\"author\\":{\\"@type\\":\\"Person\\",\\"name\\":\\"Test User\\"},\\"publisher\\":{\\"@type\\":\\"Organization\\",\\"name\\":\\"Test User\\",\\"logo\\":{\\"@type\\":\\"ImageObject\\",\\"url\\":\\"https://inspiringhopechurch.com/src/assets/logo.svg\\"}},\\"isPartOf\\":\\"https://inspiringhopechurch.com\\",\\"mainEntityOfPage\\":{\\"@type\\":\\"WebSite\\",\\"@id\\":\\"https://inspiringhopechurch.com\\"}}]",
                "type": "application/ld+json",
              },
            ]
          }
          title="Test Company Site"
        />
      </SideEffect(NullComponent)>
    </HelmetWrapper>
  </mockConstructor>
</SEO>
`;

exports[`SEO should render without crashing as an article with custom props 1`] = `
<SEO
  article={true}
  banner="/path/to/image.jpg"
  desc="This website is an awesome website and you should visit it."
  pathname="/path/to/page"
  title="Test Page"
>
  <mockConstructor
    render={[Function]}
  >
    <HelmetWrapper
      defer={false}
      encodeSpecialCharacters={true}
      title="Test Page | Test Company Site"
    >
      <SideEffect(NullComponent)
        defer={false}
        encodeSpecialCharacters={true}
        htmlAttributes={
          Object {
            "class": "has-navbar-fixed-top",
            "lang": "en",
          }
        }
        link={
          Array [
            Object {
              "href": "https://inspiringhopechurch.com/blog//path/to/page",
              "rel": "canonical",
            },
          ]
        }
        meta={
          Array [
            Object {
              "content": "This website is an awesome website and you should visit it.",
              "name": "description",
            },
            Object {
              "content": "https://inspiringhopechurch.com//path/to/image.jpg",
              "name": "image",
            },
            Object {
              "content": "TestCompany",
              "name": "apple-mobile-web-app-title",
            },
            Object {
              "content": "TestCompany",
              "name": "application-name",
            },
            Object {
              "content": "Test Page | Test Company Site",
              "property": "og:site_name",
            },
            Object {
              "content": "https://inspiringhopechurch.com/blog//path/to/page",
              "property": "og:url",
            },
            Object {
              "content": "article",
              "property": "og:type",
            },
            Object {
              "content": "Test Page | Test Company Site",
              "property": "og:title",
            },
            Object {
              "content": "This website is an awesome website and you should visit it.",
              "property": "og:description",
            },
            Object {
              "content": "https://inspiringhopechurch.com//path/to/image.jpg",
              "property": "og:image",
            },
            Object {
              "content": "summary_large_image",
              "name": "twitter:card",
            },
            Object {
              "content": "Test Page | Test Company Site",
              "name": "twitter:title",
            },
            Object {
              "content": "This website is an awesome website and you should visit it.",
              "name": "twitter:description",
            },
            Object {
              "content": "https://inspiringhopechurch.com//path/to/image.jpg",
              "name": "twitter:image",
            },
            Object {
              "content": "https://inspiringhopechurch.com/blog//path/to/page",
              "name": "twitter:url",
            },
            Object {
              "content": "@inspiringhope",
              "name": "twitter:creator",
            },
            Object {
              "content": "https://twitter.com/inspiringhope/",
              "name": "twitter:site",
            },
          ]
        }
        script={
          Array [
            Object {
              "innerHTML": "[{\\"@context\\":\\"http://schema.org\\",\\"@type\\":\\"BlogPosting\\",\\"@id\\":\\"https://inspiringhopechurch.com/blog//path/to/page\\",\\"url\\":\\"https://inspiringhopechurch.com/blog//path/to/page\\",\\"name\\":\\"Test Page\\",\\"alternateName\\":\\"Test Company Site\\",\\"headline\\":\\"Test Page\\",\\"image\\":{\\"@type\\":\\"ImageObject\\",\\"url\\":\\"https://inspiringhopechurch.com//path/to/image.jpg\\"},\\"description\\":\\"This website is an awesome website and you should visit it.\\",\\"datePublished\\":\\"2019-05-09\\",\\"dateModified\\":\\"2019-05-09\\",\\"author\\":{\\"@type\\":\\"Person\\",\\"name\\":\\"Test User\\"},\\"publisher\\":{\\"@type\\":\\"Organization\\",\\"name\\":\\"Test User\\",\\"logo\\":{\\"@type\\":\\"ImageObject\\",\\"url\\":\\"https://inspiringhopechurch.com/src/assets/logo.svg\\"}},\\"isPartOf\\":\\"https://inspiringhopechurch.com\\",\\"mainEntityOfPage\\":{\\"@type\\":\\"WebSite\\",\\"@id\\":\\"https://inspiringhopechurch.com\\"}}]",
              "type": "application/ld+json",
            },
          ]
        }
        title="Test Page | Test Company Site"
      >
        <NullComponent
          defer={false}
          encodeSpecialCharacters={true}
          htmlAttributes={
            Object {
              "class": "has-navbar-fixed-top",
              "lang": "en",
            }
          }
          link={
            Array [
              Object {
                "href": "https://inspiringhopechurch.com/blog//path/to/page",
                "rel": "canonical",
              },
            ]
          }
          meta={
            Array [
              Object {
                "content": "This website is an awesome website and you should visit it.",
                "name": "description",
              },
              Object {
                "content": "https://inspiringhopechurch.com//path/to/image.jpg",
                "name": "image",
              },
              Object {
                "content": "TestCompany",
                "name": "apple-mobile-web-app-title",
              },
              Object {
                "content": "TestCompany",
                "name": "application-name",
              },
              Object {
                "content": "Test Page | Test Company Site",
                "property": "og:site_name",
              },
              Object {
                "content": "https://inspiringhopechurch.com/blog//path/to/page",
                "property": "og:url",
              },
              Object {
                "content": "article",
                "property": "og:type",
              },
              Object {
                "content": "Test Page | Test Company Site",
                "property": "og:title",
              },
              Object {
                "content": "This website is an awesome website and you should visit it.",
                "property": "og:description",
              },
              Object {
                "content": "https://inspiringhopechurch.com//path/to/image.jpg",
                "property": "og:image",
              },
              Object {
                "content": "summary_large_image",
                "name": "twitter:card",
              },
              Object {
                "content": "Test Page | Test Company Site",
                "name": "twitter:title",
              },
              Object {
                "content": "This website is an awesome website and you should visit it.",
                "name": "twitter:description",
              },
              Object {
                "content": "https://inspiringhopechurch.com//path/to/image.jpg",
                "name": "twitter:image",
              },
              Object {
                "content": "https://inspiringhopechurch.com/blog//path/to/page",
                "name": "twitter:url",
              },
              Object {
                "content": "@inspiringhope",
                "name": "twitter:creator",
              },
              Object {
                "content": "https://twitter.com/inspiringhope/",
                "name": "twitter:site",
              },
            ]
          }
          script={
            Array [
              Object {
                "innerHTML": "[{\\"@context\\":\\"http://schema.org\\",\\"@type\\":\\"BlogPosting\\",\\"@id\\":\\"https://inspiringhopechurch.com/blog//path/to/page\\",\\"url\\":\\"https://inspiringhopechurch.com/blog//path/to/page\\",\\"name\\":\\"Test Page\\",\\"alternateName\\":\\"Test Company Site\\",\\"headline\\":\\"Test Page\\",\\"image\\":{\\"@type\\":\\"ImageObject\\",\\"url\\":\\"https://inspiringhopechurch.com//path/to/image.jpg\\"},\\"description\\":\\"This website is an awesome website and you should visit it.\\",\\"datePublished\\":\\"2019-05-09\\",\\"dateModified\\":\\"2019-05-09\\",\\"author\\":{\\"@type\\":\\"Person\\",\\"name\\":\\"Test User\\"},\\"publisher\\":{\\"@type\\":\\"Organization\\",\\"name\\":\\"Test User\\",\\"logo\\":{\\"@type\\":\\"ImageObject\\",\\"url\\":\\"https://inspiringhopechurch.com/src/assets/logo.svg\\"}},\\"isPartOf\\":\\"https://inspiringhopechurch.com\\",\\"mainEntityOfPage\\":{\\"@type\\":\\"WebSite\\",\\"@id\\":\\"https://inspiringhopechurch.com\\"}}]",
                "type": "application/ld+json",
              },
            ]
          }
          title="Test Page | Test Company Site"
        />
      </SideEffect(NullComponent)>
    </HelmetWrapper>
  </mockConstructor>
</SEO>
`;

exports[`SEO should render without crashing with custom props 1`] = `
<SEO
  article={false}
  banner="/path/to/image.jpg"
  desc="This website is an awesome website and you should visit it."
  pathname="/path/to/page"
  title="Test Page"
>
  <mockConstructor
    render={[Function]}
  >
    <HelmetWrapper
      defer={false}
      encodeSpecialCharacters={true}
      title="Test Page | Test Company Site"
    >
      <SideEffect(NullComponent)
        defer={false}
        encodeSpecialCharacters={true}
        htmlAttributes={
          Object {
            "class": "has-navbar-fixed-top",
            "lang": "en",
          }
        }
        link={
          Array [
            Object {
              "href": "https://inspiringhopechurch.com//path/to/page",
              "rel": "canonical",
            },
          ]
        }
        meta={
          Array [
            Object {
              "content": "This website is an awesome website and you should visit it.",
              "name": "description",
            },
            Object {
              "content": "https://inspiringhopechurch.com//path/to/image.jpg",
              "name": "image",
            },
            Object {
              "content": "TestCompany",
              "name": "apple-mobile-web-app-title",
            },
            Object {
              "content": "TestCompany",
              "name": "application-name",
            },
            Object {
              "content": "Test Page | Test Company Site",
              "property": "og:site_name",
            },
            Object {
              "content": "https://inspiringhopechurch.com//path/to/page",
              "property": "og:url",
            },
            Object {
              "content": "website",
              "property": "og:type",
            },
            Object {
              "content": "Test Page | Test Company Site",
              "property": "og:title",
            },
            Object {
              "content": "This website is an awesome website and you should visit it.",
              "property": "og:description",
            },
            Object {
              "content": "https://inspiringhopechurch.com//path/to/image.jpg",
              "property": "og:image",
            },
            Object {
              "content": "summary_large_image",
              "name": "twitter:card",
            },
            Object {
              "content": "Test Page | Test Company Site",
              "name": "twitter:title",
            },
            Object {
              "content": "This website is an awesome website and you should visit it.",
              "name": "twitter:description",
            },
            Object {
              "content": "https://inspiringhopechurch.com//path/to/image.jpg",
              "name": "twitter:image",
            },
            Object {
              "content": "https://inspiringhopechurch.com//path/to/page",
              "name": "twitter:url",
            },
            Object {
              "content": "@inspiringhope",
              "name": "twitter:creator",
            },
            Object {
              "content": "https://twitter.com/inspiringhope/",
              "name": "twitter:site",
            },
          ]
        }
        script={
          Array [
            Object {
              "innerHTML": "[{\\"@context\\":\\"http://schema.org\\",\\"@type\\":\\"WebSite\\",\\"@id\\":\\"https://inspiringhopechurch.com\\",\\"url\\":\\"https://inspiringhopechurch.com\\",\\"name\\":\\"Test Company Site\\",\\"alternateName\\":\\"Test Company Site\\"}]",
              "type": "application/ld+json",
            },
          ]
        }
        title="Test Page | Test Company Site"
      >
        <NullComponent
          defer={false}
          encodeSpecialCharacters={true}
          htmlAttributes={
            Object {
              "class": "has-navbar-fixed-top",
              "lang": "en",
            }
          }
          link={
            Array [
              Object {
                "href": "https://inspiringhopechurch.com//path/to/page",
                "rel": "canonical",
              },
            ]
          }
          meta={
            Array [
              Object {
                "content": "This website is an awesome website and you should visit it.",
                "name": "description",
              },
              Object {
                "content": "https://inspiringhopechurch.com//path/to/image.jpg",
                "name": "image",
              },
              Object {
                "content": "TestCompany",
                "name": "apple-mobile-web-app-title",
              },
              Object {
                "content": "TestCompany",
                "name": "application-name",
              },
              Object {
                "content": "Test Page | Test Company Site",
                "property": "og:site_name",
              },
              Object {
                "content": "https://inspiringhopechurch.com//path/to/page",
                "property": "og:url",
              },
              Object {
                "content": "website",
                "property": "og:type",
              },
              Object {
                "content": "Test Page | Test Company Site",
                "property": "og:title",
              },
              Object {
                "content": "This website is an awesome website and you should visit it.",
                "property": "og:description",
              },
              Object {
                "content": "https://inspiringhopechurch.com//path/to/image.jpg",
                "property": "og:image",
              },
              Object {
                "content": "summary_large_image",
                "name": "twitter:card",
              },
              Object {
                "content": "Test Page | Test Company Site",
                "name": "twitter:title",
              },
              Object {
                "content": "This website is an awesome website and you should visit it.",
                "name": "twitter:description",
              },
              Object {
                "content": "https://inspiringhopechurch.com//path/to/image.jpg",
                "name": "twitter:image",
              },
              Object {
                "content": "https://inspiringhopechurch.com//path/to/page",
                "name": "twitter:url",
              },
              Object {
                "content": "@inspiringhope",
                "name": "twitter:creator",
              },
              Object {
                "content": "https://twitter.com/inspiringhope/",
                "name": "twitter:site",
              },
            ]
          }
          script={
            Array [
              Object {
                "innerHTML": "[{\\"@context\\":\\"http://schema.org\\",\\"@type\\":\\"WebSite\\",\\"@id\\":\\"https://inspiringhopechurch.com\\",\\"url\\":\\"https://inspiringhopechurch.com\\",\\"name\\":\\"Test Company Site\\",\\"alternateName\\":\\"Test Company Site\\"}]",
                "type": "application/ld+json",
              },
            ]
          }
          title="Test Page | Test Company Site"
        />
      </SideEffect(NullComponent)>
    </HelmetWrapper>
  </mockConstructor>
</SEO>
`;
